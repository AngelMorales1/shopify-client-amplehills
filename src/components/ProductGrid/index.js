import React from 'react';
import PropTypes from 'prop-types';
import productModel from 'models/productModel';
import ProductGridCard from 'components/ProductGridCard';

const ProductGrid = ({ products }) => {
  return (
    <div className="ProductGrid">
      <div className="flex flex-wrap container-width mx-auto px2">
        {Object.values(products).map(product => (
          <ProductGridCard key={product.id} product={product} />
        ))}
      </div>
    </div>
  );
};

export default ProductGrid;

ProductGrid.propTypes = {
  products: PropTypes.objectOf(productModel.propTypes)
};

ProductGrid.defaultProps = {
  products: {}
};
