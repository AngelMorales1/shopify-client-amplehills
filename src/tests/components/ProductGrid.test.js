import React from 'react';
import { shallow } from 'enzyme';

import ProductGrid from 'components/ProductGrid';

it('renders without data', () => {
  const component = shallow(<ProductGrid />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <ProductGrid
      products={{
        foo: {
          vailable: true,
          blocks: [{}],
          cartDetails: 'foo',
          description: 'foo',
          flavorDescription: 'foo',
          gridImage: 'foo',
          handle: 'foo',
          id: 'foo',
          image: 'foo',
          pintImage: 'foo',
          price: 0,
          subItems: [],
          subItemsAvailable: true,
          title: 'foo'
        }
      }}
    />
  );

  expect(component).toMatchSnapshot();
});
