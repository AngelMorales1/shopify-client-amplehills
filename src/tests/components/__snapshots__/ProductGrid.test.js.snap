// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductGrid
    merchandise={false}
    products={
        Array [
            Object {
              "available": true,
              "blocks": Array [
                Object {},
              ],
              "cartDetails": "foo",
              "description": "foo",
              "flavorDescription": "foo",
              "gridImage": "foo",
              "handle": "foo",
              "id": "foo",
              "image": "foo",
              "pintImage": "foo",
              "price": 0,
              "subItems": Array [],
              "subItemsAvailable": true,
              "title": "foo",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex flex-wrap container-width mx-auto px2 justify-center"
>
        <ProductGridCard
                product={
                        Object {
                                "available": true,
                                "blocks": Array [
                                  Object {},
                                ],
                                "cartDetails": "foo",
                                "description": "foo",
                                "flavorDescription": "foo",
                                "gridImage": "foo",
                                "handle": "foo",
                                "id": "foo",
                                "image": "foo",
                                "pintImage": "foo",
                                "price": 0,
                                "subItems": Array [],
                                "subItemsAvailable": true,
                                "title": "foo",
                              }
                }
                productIsMerchandise={false}
        />
</div>,
      "className": "ProductGrid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProductGridCard
            product={
                        Object {
                                    "available": true,
                                    "blocks": Array [
                                      Object {},
                                    ],
                                    "cartDetails": "foo",
                                    "description": "foo",
                                    "flavorDescription": "foo",
                                    "gridImage": "foo",
                                    "handle": "foo",
                                    "id": "foo",
                                    "image": "foo",
                                    "pintImage": "foo",
                                    "price": 0,
                                    "subItems": Array [],
                                    "subItemsAvailable": true,
                                    "title": "foo",
                                  }
            }
            productIsMerchandise={false}
/>,
        ],
        "className": "flex flex-wrap container-width mx-auto px2 justify-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "foo",
          "nodeType": "function",
          "props": Object {
            "product": Object {
              "available": true,
              "blocks": Array [
                Object {},
              ],
              "cartDetails": "foo",
              "description": "foo",
              "flavorDescription": "foo",
              "gridImage": "foo",
              "handle": "foo",
              "id": "foo",
              "image": "foo",
              "pintImage": "foo",
              "price": 0,
              "subItems": Array [],
              "subItemsAvailable": true,
              "title": "foo",
            },
            "productIsMerchandise": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex flex-wrap container-width mx-auto px2 justify-center"
>
          <ProductGridCard
                    product={
                              Object {
                                        "available": true,
                                        "blocks": Array [
                                          Object {},
                                        ],
                                        "cartDetails": "foo",
                                        "description": "foo",
                                        "flavorDescription": "foo",
                                        "gridImage": "foo",
                                        "handle": "foo",
                                        "id": "foo",
                                        "image": "foo",
                                        "pintImage": "foo",
                                        "price": 0,
                                        "subItems": Array [],
                                        "subItemsAvailable": true,
                                        "title": "foo",
                                      }
                    }
                    productIsMerchandise={false}
          />
</div>,
        "className": "ProductGrid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ProductGridCard
              product={
                            Object {
                                          "available": true,
                                          "blocks": Array [
                                            Object {},
                                          ],
                                          "cartDetails": "foo",
                                          "description": "foo",
                                          "flavorDescription": "foo",
                                          "gridImage": "foo",
                                          "handle": "foo",
                                          "id": "foo",
                                          "image": "foo",
                                          "pintImage": "foo",
                                          "price": 0,
                                          "subItems": Array [],
                                          "subItemsAvailable": true,
                                          "title": "foo",
                                        }
              }
              productIsMerchandise={false}
/>,
          ],
          "className": "flex flex-wrap container-width mx-auto px2 justify-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "foo",
            "nodeType": "function",
            "props": Object {
              "product": Object {
                "available": true,
                "blocks": Array [
                  Object {},
                ],
                "cartDetails": "foo",
                "description": "foo",
                "flavorDescription": "foo",
                "gridImage": "foo",
                "handle": "foo",
                "id": "foo",
                "image": "foo",
                "pintImage": "foo",
                "price": 0,
                "subItems": Array [],
                "subItemsAvailable": true,
                "title": "foo",
              },
              "productIsMerchandise": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductGrid
    merchandise={false}
    products={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex flex-wrap container-width mx-auto px2 justify-center"
>
        
</div>,
      "className": "ProductGrid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "flex flex-wrap container-width mx-auto px2 justify-center",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex flex-wrap container-width mx-auto px2 justify-center"
>
          
</div>,
        "className": "ProductGrid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "flex flex-wrap container-width mx-auto px2 justify-center",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
